src/server/transaction.cc

```diff
@@ -157,38 +157,23 @@ void Transaction::InitShardData(absl::Span<const PerShardCache> shard_index, siz
   CHECK_EQ(args_.size(), num_args);
 }
 
-void Transaction::InitMultiData(KeyIndex key_index) {
+void Transaction::RecordMultiLocks(const KeyIndex& key_index) {
   DCHECK(multi_);
+  DCHECK(!multi_->lock_mode);
 
   if (multi_->mode == NON_ATOMIC)
     return;
 
-  IntentLock::Mode mode = Mode();
-
-  auto& tmp_uniques = tmp_space.uniq_keys;
-
-  auto lock_key = [this, mode, &tmp_uniques](string_view key) {
-    if (auto [_, inserted] = tmp_uniques.insert(KeyLockArgs::GetLockKey(key)); !inserted)
-      return;
-
-    multi_->lock_counts[key][mode]++;
-  };
+  auto lock_key = [this](string_view key) { multi_->locks.emplace(KeyLockArgs::GetLockKey(key)); };
```

Comment:
@dranikpg please review this change. I have not found the use-case where we lock the same key several times for the same transaction. Could be the result of dropping the incremental locking support.

Reply:
Yes, it is. We actually don't need to track lock counts